crates/binjs_io/src/entropy/write/mod.rs

```diff
@@ -308,9 +308,30 @@ impl Encoder {
 
         // Write (and possibly dump) data.
         // In the current implementation, we just ignore any information other than the index.
+
+        // We reserve number 0 for "next in prelude", as it is a very common pattern, and having
+        // successive 0s improves compression (confirmed by issue #298).
+        // Any other index i is written as i + 1.
+        let mut latest_in_prelude = None;
         for item in vec {
-            let index = item.raw();
-            lazy_stream.write_varnum(index as u32)?;
+            let varnum = match *item {
+                // References to the shared dictionary are shifted by 1.
+                Index::Shared(i) => i + 1,
+                // References to the prelude dictionary may be 0 (for next) or shifted by 1.
```

Comment:
Please move the comment and code about the 0/shifting to `Index` enum's declaration/implementation,
and not to embed in random place.
both encoding and decoding.